File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/dtos/OrderDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.entities.BaseDTO;
import com.project.shopapp.entities.BaseEntity;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Data
@Getter
@Builder
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OrderDTO  {
    @Min(value = 1, message = "User id must be > 0")
    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("fullname")
    private String fullName;

    private String email;


    @JsonProperty("phone_number")
    @Size(min = 5, message = "Phone number must be at least 5 characters")
    private String phoneNumber;

    private String address;

    @JsonProperty("total_money")
    @Min(value = 0, message = "Total money must be >= 0")
    private float totalMoney;

    private String note;

    @JsonProperty("shipping_method")
    @NotBlank(message = "Shipping method is required")
    private String shippingMethod;

    @JsonProperty("shipping_date")
    private LocalDate shippingDate;

    @JsonProperty("payment_method")
    @NotBlank(message = "Payment method is required")
    private String paymentMethod;

    @JsonProperty("cart_items")
    @NotEmpty(message = "Item can't empty")
    private List<CartItemDTO> cartItemDTOS;
}

File: shopapp-angular/src/app/dtos/order/order.dto.ts
import { CartItem } from './cart.item.dtos';

export class OrderDTO {
  user_id: number;
  full_name: string;
  email: string;
  phone_number: string;
  address: string;
  note: string;
  total_money: number;
  payment_method: string;
  shipping_method: string;
  coupon_code: string;
  cart_items: CartItem[];
  constructor(data: any) {
    this.full_name = data.fullname;
    this.user_id = data.user_id;
    this.email = data.email;
    this.address = data.address;
    this.payment_method = data.payment_method;
    this.phone_number = data.phone_number;
    this.note = data.note;
    this.total_money = data.total_money;
    this.shipping_method = data.shippng_method;
    this.coupon_code = data.coupon_code;
    this.cart_items = data.cart_items;
  }
}
